### PR TITLE
Spelling silgen

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift silgen, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101092698

### CODE REVIEW

File: lib/SILGen/SILGenType.cpp [hunk 1]
```diff
@@ -477,9 +477,9 @@ template<typename T> class SILGenWitnessTable : public SILWitnessVisitor<T> {
     // substitute the witness's derivative generic signature in its derivative
     // function identifier.
     if (requirementRef.isAutoDiffDerivativeFunction()) {
-      auto *reqrRerivativeId = requirementRef.getDerivativeFunctionIdentifier();
+      auto *reqrDerivativeId = requirementRef.getDerivativeFunctionIdentifier();
```
Comment (jsoref) [inline lib/SILGen/SILGenType.cpp:480]: notable

File: test/SILGen/dynamically_replaceable.swift [hunk 2]
```diff
@@ -211,34 +211,34 @@ extension Klass {
   }
 }
 
-extension Strukt {
+extension Struct {
```
Comment (jsoref) [inline test/SILGen/dynamically_replaceable.swift:214]: It's possible this is a reserved word...

File: test/SILGen/hop_to_executor_async_prop.swift [hunk 3]
```diff
@@ -648,22 +648,22 @@ struct StateObject<ObjectType> {
...
 }
 
-final private actor Coordinactor {
+final private actor Coordinator {
     var someValue: Int?
 }
 
 struct Blah {
-    @StateObject private var coordinator = Coordinactor()
+    @StateObject private var coordinator = Coordinator()
```
Comment (jsoref) [inline test/SILGen/hop_to_executor_async_prop.swift:656]: I can't tell if this is an intentional cross or a typo.

File: test/SILGen/dynamically_replaceable.swift (hunk 2 above)
Comment (tshortli) [reply test/SILGen/dynamically_replaceable.swift:214]: `Struct` with an uppercase `S` appears to not be reserved. However, given the class decl `Klass` above I think the spelling here seems intentional enough to me that I would drop this from the PR.

File: test/SILGen/hop_to_executor_async_prop.swift (hunk 3 above)
Comment (tshortli) [reply test/SILGen/hop_to_executor_async_prop.swift:656]: @kavon is this an intentional spelling you'd like to keep in the test?

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by jsoref** [comment]
@tshortli: do I need to rebase/do something here?

**Comment by tshortli** [comment]
> @tshortli: do I need to rebase/do something here?

Sorry for losing track of this PR. I'll run the tests and assuming there aren't any issues we can rebase and squash these changes.

**Comment by tshortli** [comment]
@swift-ci please smoke test

File: test/SILGen/dynamically_replaceable.swift (hunk 2 above)
Comment (jsoref) [reply test/SILGen/dynamically_replaceable.swift:214]: Hmm, looks like I failed to drop this. I can...

**Comment by tshortli** [comment]
@jsoref omitting the changes affecting these tests might be the most expedient way to get the rest of this landed:
```
Failed Tests (2):
  Swift(macosx-x86_64) :: SILGen/objc_disable_brigding.swift
  Swift(macosx-x86_64) :: SILGen/specialize_attr.swift
```

**Comment by jsoref** [comment]
@tshortli:  works for me.

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test macOS

**Comment by tshortli** [comment]
@jsoref looks good to squash when you're ready

**Comment by tshortli** [comment]
@swift-ci please smoke test and merge

**Review by tshortli — APPROVED** [review]

**Pull request merged**